src/vs/workbench/api/common/extHostNotebook.ts

```diff
@@ -796,6 +803,36 @@ export class ExtHostNotebookController implements ExtHostNotebookShape {
 
 		return statusBarItem;
 	}
+
+	createNotebookCellExecution(docUri: vscode.Uri, index: number, kernelId: string): vscode.NotebookCellExecutionTask | undefined {
+		const document = this.lookupNotebookDocument(docUri);
+		if (!document) {
+			throw new Error(`Invalid cell uri/index: ${docUri}, ${index}`);
+		}
+
+		const cell = document.getCellByIndex(index);
+		if (!cell) {
+			throw new Error(`Invalid cell uri/index: ${docUri}, ${index}`);
+		}
+
+		// TODO@roblou more to do here?
+		// TODO@roblou also validate kernelId, once kernel has moved from editor to document
+		if (this._activeExecutions.has(cell.uri)) {
+			return;
```

Comment:
Unsure if this should be prohibited and iff so you must also check on the renderer because there are N extension hosts

Reply:
If only the active kernel is allowed to have executions, then I think this is sufficient. That isn't implemented but I think we want that.